backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -141,88 +151,109 @@ public User Get(Guid id)
         /// <summary>
         /// Add the specified user to the datasource.
         /// </summary>
-        /// <param name="entity"></param>
+        /// <param name="user"></param>
         /// <returns></returns>
-        public override void Add(User entity)
+        public override void Add(User user)
         {
-            entity.ThrowIfNull(nameof(entity));
+            user.ThrowIfNull(nameof(user));
 
-            entity.Roles.ForEach(r => this.Context.Entry(r).State = EntityState.Added);
-            entity.Agencies.ForEach(a => this.Context.Entry(a).State = EntityState.Added);
+            user.Roles.ForEach(r => this.Context.Entry(r).State = EntityState.Added);
+            user.Agencies.ForEach(a => this.Context.Entry(a).State = EntityState.Added);
 
-            base.Add(entity);
-            this.Context.Detach(entity);
+            base.Add(user);
+            this.Context.Detach(user);
         }
 
         /// <summary>
         /// Updates the specified user in the datasource.
         /// </summary>
-        /// <param name="entity"></param>
+        /// <param name="user"></param>
         /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
         /// <returns></returns>
-        public override void Update(User entity)
+        public override void Update(User user)
         {
-            entity.ThrowIfNull(nameof(entity));
+            user.ThrowIfNull(nameof(user));
 
-            var user = this.Context.Users
+            var existingUser = this.Context.Users
                 .Include(u => u.Agencies)
                 .Include(u => u.Roles)
                 .AsNoTracking()
-                .FirstOrDefault(u => u.Id == entity.Id) ?? throw new KeyNotFoundException();
+                .FirstOrDefault(u => u.Id == user.Id) ?? throw new KeyNotFoundException();
 
-            this.Context.SetOriginalRowVersion(user);
+            this.Context.SetOriginalRowVersion(existingUser);
 
-            var addRoles = entity.Roles.Except(user.Roles, new UserRoleRoleIdComparer());
+            if (!existingUser.Agencies.Any())
```

Comment:
If a user is updated directly without approving their access request, capture who and when.